### PR TITLE
Docker!

### PR DESCRIPTION
**Pull request by kevinahuber**
This adds a Dockerfile and rearranges some of the npm scripts to better accommodate and follow best practices.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,19 +38,19 @@
...
+    "build": "npm run build:css && react-scripts build",
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject",
-    "lint": "yarn lint:js",
+    "lint": "npm run lint:js",
     "lint:js": "eslint .",
-    "serve": "NODE_ENV=development node server/index.js",
-    "deploy": "NODE_ENV=production node server/index.js"
+    "dev:server": "NODE_ENV=development node server/index.js",
+    "start": "NODE_ENV=production node server/index.js"
```
Comment (kevinahuber) [inline package.json]: `start` is the default "start server" command, so I swapped this up to reflect that. I also didn't want to mess with installing yarn on the docker container... although we will want to do that or switch to an npm lockfile in the near future.

File: package.json [hunk 2]
```diff
@@ -38,19 +38,19 @@
...
-    "build-css":
+    "build:css":
       "node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/",
-    "watch-css":
-      "yarn build-css && node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/ --watch --recursive",
-    "start-js": "react-scripts start",
-    "start": "npm-run-all -p watch-css start-js",
-    "build": "yarn build-css && react-scripts build",
+    "watch:css":
+      "npm run build-css && node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/ --watch --recursive",
```
Comment (cereallarceny) [inline package.json]: Should be `build:css`

File: package.json (hunk 1 above)
Comment (cereallarceny) [reply package.json]: That's true, but I think we should actually have the default `start` task circumnavigate SSR.  Is that necessary for local development?  Perhaps it is, but technically it just makes previewing changes a bit slower.  Can we differentiate between running non-SSR locally, and running SSR locally and in production?  Does that make sense?

Comment (kevinahuber) [reply package.json]: This does not make sense to me- please explain more.

Comment (cereallarceny) [reply package.json]: I'd love to have a way to do non-ssr locally.  Do we still have such a command or is it always running a node instance?  On the other hand, maybe we want it to... the user-agent detection could sniff this out and just forward you on to the app without "server rendering" at all.  Does that make more sense?

**Pull request closed without merging**